academy/3-ibc/4-light-client-dev.md

```diff
@@ -0,0 +1,177 @@
+
+* Why we need extra light clients being developed
+* What light client developers need to do and where to find documentation
+* How to get a new light client on your chain
+
+</HighlightBox>
+
+## IBC ecosystem expansion
+
+IBC was envisioned from the original Cosmos whitepaper to be a crucial component of the appchain thesis for the Cosmos network on the one hand, but more importantly also remain as universal as possible to enable it to become a universal standard with respect to blockchain interoperability.
+
+Whereas the protocol always envisioned the wider adoption of IBC, the ibc-go implementation initially was focused on its use within Cosmos SDK chains connecting to similar chains. The [ibc-go v7 release](https://github.com/cosmos/ibc-go/releases/tag/v7.0.0) included a refactor to the `02-client` submodule, which should streamline the development of light clients to connect to chains with other consensus than `07-tendermint`, to ibc-go.
+
+<HighlightBox type="docs">
+
+In this section, it's assumed that the version of ibc-go is >v7. For more information about the changes and the motivation, please refer to this [blog post](https://medium.com/the-interchain-foundation/client-refactor-laying-the-groundwork-for-ibc-to-expand-across-ecosystems-61ec5a1b63bc) or [ADR-006](https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-006-02-client-refactor.md) on the topic of the client refactor.
+
+</HighlightBox>
+
+## Light client development high-level overview
+
+<HighlightBox type="reading">
+
+The development of a light client for heterogeneous chains is a complex topic and outside of the scope presented here. Light client developers are urged to instead refer to [the light client developer guide](https://ibc.cosmos.network/main/ibc/light-clients/overview.html) in the ibc-go documentation.
+
+</HighlightBox>
+
+### Recap: what does a light client do?
+
+A short and succinct **summary of a light client's functionality** is the following: a light client stores a trusted consensus state and provides functionality to verify updates to the consensus state or verify packet commitments against the trusted root by using Merkle proofs.
+
+### Major interfaces
+
+Access to IBC light clients are gated by the core IBC `MsgServer` which utilizes the abstractions set by the `02-client` submodule to call into a light client module. A light client module developer is only required to implement a set interfaces as defined in the `modules/core/exported` package of ibc-go.
+
+A light client module developer should be concerned with three main interfaces:
+
+* [`ClientState`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/exported/client.go#L36) contains all the information needed to verify a `ClientMessage` and perform membership and non-membership proof verification of counterparty state. This includes properties that refer to the remote state machine, the light client type and the specific light client instance.
```

Comment:
Maybe use permalinks here? It future proofs these references.

Reply:
v7.0.0 is a tag and we can assume it will be permanent.